File: 11-container-with-most-water/container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int i = 0;
        int n = height.size();
        int j = n-1;
        int g_max = 0;
        while(i<j)
        {
            int curr = min(height[i],height[j]) * (j-i);
            g_max = max(curr, g_max);
            if(height[i] < height[j])
            {
                i++;
            }
            else
            {
                j--;
            }
        }
        return g_max;
    }
};

File: 11-container-with-most-water/container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        i, j = 0, len(height) - 1
        res = float("-inf")
        while i <= j:
            current = (j-i) * min(height[i], height[j])
            res = max(current, res)
            if height[i] < height[j]:
                i+=1
            elif height[i] > height[j]:
                j-=1
            elif height[i] == height[j]:
                i+=1
                j-=1
        return res